### PR TITLE
Document syntax for accessing nested parameters in `change_format` - `when` tags

### PR DESCRIPTION
**Pull request by bernt-matthias**
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 1]
```diff
@@ -7394,13 +7394,16 @@ or the test tool
 [output_format.xml](https://github.com/galaxyproject/galaxy/blob/dev/test/functional/tools/output_format.xml)
 for more examples.
 
+For parameters that are nested in sections, conditionals, or repeats are accessed with dictionary access syntax,
+e.g. a parameter with name ``p`` that is in a conditinal with name ``c``that is in a section with name ``s``
+is referenced by ``s["c"]["p"].
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
For parameters that are nested in sections, conditionals, or repeats are accessed with object access syntax,
e.g. a parameter with name ``p`` that is in a conditional with name ``c``that is in a section with name ``s``
is referenced by ``s.c.p``"].
```

Comment (mvdbeek) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: Not sure where you've found the dictionary access form, IUC is mostly (or exclusively from a quick glance) using `outer.inner`

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 2]
```diff
@@ -7394,13 +7394,16 @@ or the test tool
...
 ]]></xs:documentation>
     </xs:annotation>
     <xs:sequence/>
     <xs:attribute name="input" type="xs:string">
       <xs:annotation>
         <xs:documentation xml:lang="en">This attribute should be the name of
-the desired input parameter (e.g. ``input="out_format"`` above).</xs:documentation>
+the desired input parameter (e.g. ``input="out_format"`` above). Parameters that are nested are accessed like a dictionarty.</xs:documentation>
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
the desired input parameter (e.g. ``input="out_format"`` above). Parameters that are nested are accessed like an object.</xs:documentation>
```

File: lib/galaxy/tool_util/xsd/galaxy.xsd (hunk 1 above)
Comment (bernt-matthias) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: Works definitely as `s["c"]["p"]` just used it it https://github.com/galaxyproject/tools-iuc/pull/5956 and also double checked with a few print statements in the code (but not sure if `s.c.p` does not work. Code is [here](https://github.com/galaxyproject/galaxy/blob/ffc66f2c28175f99cedfd50998696fb38461f402/lib/galaxy/tools/actions/__init__.py#L1232) where `parameter_context` is a `dict`).

Comment (bernt-matthias) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: Lets try https://github.com/galaxyproject/tools-iuc/pull/5956/commits/8cb7361a3a9a392e03896f8c9a57c6ca706884e6 :)

Comment (bernt-matthias) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: Seems to work both. But lets go with the consensus. Thanks.

**Pull request merged**